src/components/link/link.tsx

```diff
@@ -37,14 +36,13 @@ export interface LinkButtonProps {
   type?: EuiLinkType;
   color?: EuiLinkColor;
   onClick?: MouseEventHandler<HTMLButtonElement>;
-}
-
-export type EuiLinkButtonProps = {
   /**
    * Set this prop to true if you want to show an icon indicating that this is an external link.
    */
   external?: boolean;
```

Comment:
Instead of adding this directly on `EuiLinkButtonProps`, this should be part of the `LinkButtonProps` interface.
This is confusing, for sure. Think of it as an additive approach where the last defined thing wins. In this case, we want all of the custom props to go last, as part of `LinkButtonProps`